packages/data-explorer/src/components/ColumnSelection.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <div class="d-flex flex-column">
+    <label v-for="column in allColumns" :key="column"
+      ><input
+        class="ml-2"
+        type="checkbox"
+        :value="column"
+        @change="modifyHiddenColumns($event)"
+        :checked="!hiddenColumns.includes(column)"
+      />
+      <span class="ml-2" v-text="column"></span>
+    </label>
+  </div>
+</template>
+
+<script>
+export default {
+  computed: {
+    allColumns () {
+      return this.$store.state.tableData
```

Comment:
general architecture is to not use the store in the components, but props down and events up, then use the components in the Views and use the View to wire it to the store

Reply:
then we need to have more code in the component, to watch for changes. But can be done